jschon/jsonschema.py

```diff
@@ -137,6 +137,19 @@ def _bootstrap(self, value: Mapping[str, JSONCompatible]) -> None:
                 self.keywords[key] = kw
                 self.data[key] = kw.json
 
+        if '$id' in value:
+            if str(self.metaschema.core_vocabulary.uri) in (
+                'https://json-schema.org/draft/2019-09/vocab/core',
+                'https://json-schema.org/draft/2020-12/vocab/core',
+            ):
+                from jschon.vocabulary.core import IdKeyword_2019_09 as IdKeyword
+            else:
+                from jschon.vocabulary.core import IdKeyword_Next as IdKeyword
+
+            id_kw = IdKeyword(self, value['$id'])
+            self.keywords['$id'] = id_kw
+            self.data['$id'] = id_kw.json
+
```

Comment:
All the string values above should be double-quoted. I'm fairly certain that I've not been consistent with this across the codebase, but in general I'm following a convention of double-quoting any string that comes from a schema or meta-schema JSON document, and (going forward) single-quoting everything else.